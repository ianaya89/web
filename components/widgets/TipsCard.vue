<template>
  <div class="tips-card py-6 px-5 min-h-64" data-cy="tipsCard">
    <h4 class="font-medium uppercase my-3">
      {{ count }} {{ $t('widgets.tips.title') }}
    </h4>
    <div class="flex justify-between items-center my-3">
      <span class="font-medium text-h6 py-2">
        {{ $t('widgets.tips.label') }} {{ tipIndex + 1 }}
      </span>
      <div class="select-none font-bold flex justify-end">
        <a
          :class="{disabled : tipIndex === 0}"
          class="lni-chevron-left text-red hover:text-red-d1 px-2 py-2 font-bold cursor-pointer"
          @click="previous"
        />
        <a
          :class="{disabled : count === 0 || count - 1 === tipIndex}"
          class="lni-chevron-right text-red hover:text-red-d1 px-2 py-2 font-bold cursor-pointer"
          @click="next"
        />
      </div>
    </div>
    <div v-if="$apollo.loading">
      <p>
        <content-placeholders>
          <content-placeholders-text :lines="3" />
        </content-placeholders>
      </p>
    </div>
    <div
      v-if="!$apollo.loading && count !== 0"
      class="font-serif text-menu text-grey"
    >
      <transition name="slide-fade" appear mode="out-in">
        <p :key="tipIndex" v-text="body" />
      </transition>
    </div>

    <div v-if="!$apollo.loading && count === 0">
      {{ $t('widgets.tips.no_tips') }}
    </div>
  </div>
</template>

<script>
import { TipsQuery } from '@/services/graphql'

export default {
  name: 'TipsCard',

  apollo: {
    tips: TipsQuery
  },

  data: () => ({
    tipIndex: 0,
    tips: TipsQuery.mock
  }),

  computed: {
    count() {
      return this.tips.length
    },

    body() {
      return this.$get(
        this.tips,
        `${this.tipIndex}.body`,
        this.$t('widgets.tips.empty_body')
      )
    }
  },

  methods: {
    next() {
      if (this.tipIndex < this.count - 1) {
        this.tipIndex++
      }
    },

    previous() {
      if (this.tipIndex !== 0) {
        this.tipIndex--
      }
    }
  }
}
</script>

<style lang="postcss" scoped>
.tips-card {
  a.disabled {
    color: grey !important;
    cursor: default;
    &:hover {
      color: grey !important;
    }
  }
}
</style>
