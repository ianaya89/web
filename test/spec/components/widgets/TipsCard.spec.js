import { mount, shallowMount } from '@vue/test-utils'
import TipsCard from '@/components/widgets/TipsCard'

describe('TipsCard', () => {
  it('renders TipsCard', () => {
    const wrapper = shallowMount(TipsCard)

    expect(wrapper.element).toMatchSnapshot()
  })

  it('has correct number of tips', () => {
    const wrapper = mount(TipsCard)

    wrapper.vm.tips = [1, 2]

    expect(wrapper.vm.count).toEqual(2)
  })

  it('has correct body text', () => {
    const wrapper = mount(TipsCard)

    wrapper.vm.tips = [
      {
        body: 'lorem ipsum'
      }
    ]

    expect(wrapper.vm.body).toEqual('lorem ipsum')
  })

  it('can move tipIndex using methods', () => {
    const wrapper = mount(TipsCard)

    wrapper.vm.tips = [1, 2]

    expect(wrapper.vm.tipIndex).toEqual(0)

    wrapper.vm.next()
    expect(wrapper.vm.tipIndex).toEqual(1)

    wrapper.vm.next()
    expect(wrapper.vm.tipIndex).toEqual(1)

    wrapper.vm.previous()
    expect(wrapper.vm.tipIndex).toEqual(0)

    wrapper.vm.previous()
    expect(wrapper.vm.tipIndex).toEqual(0)
  })
})
