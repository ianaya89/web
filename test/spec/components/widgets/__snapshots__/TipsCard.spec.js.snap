// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TipsCard renders TipsCard 1`] = `
<div
  class="tips-card py-6 px-5 min-h-64"
  data-cy="tipsCard"
>
  <h4
    class="font-medium uppercase my-3"
  >
    
    0 TRANSLATE:widgets.tips.title
  
  </h4>
   
  <div
    class="flex justify-between items-center my-3"
  >
    <span
      class="font-medium text-h6 py-2"
    >
      
      TRANSLATE:widgets.tips.label 1
    
    </span>
     
    <div
      class="select-none font-bold flex justify-end"
    >
      <a
        class="lni-chevron-left text-red hover:text-red-d1 px-2 py-2 font-bold cursor-pointer disabled"
      />
       
      <a
        class="lni-chevron-right text-red hover:text-red-d1 px-2 py-2 font-bold cursor-pointer disabled"
      />
    </div>
  </div>
   
  <!---->
   
  <!---->
   
  <div>
    
    TRANSLATE:widgets.tips.no_tips
  
  </div>
</div>
`;
